tracer/src/Datadog.Trace/ClrProfiler/NativeCallTargetDefinition2.cs

```diff
@@ -0,0 +1,157 @@
+// <copyright file="NativeCallTargetDefinition2.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Datadog.Trace.ClrProfiler
+{
+ // !                                         ██
+ //                                         ██░░██
+ //                                       ██░░░░░░██
+ //                                     ██░░░░░░░░░░██
+ //                                     ██░░░░░░░░░░██
+ //                                   ██░░░░░░░░░░░░░░██
+ //                                 ██░░░░░░██████░░░░░░██
+ //                                 ██░░░░░░██████░░░░░░██
+ //                               ██░░░░░░░░██████░░░░░░░░██
+ //                               ██░░░░░░░░██████░░░░░░░░██
+ //                             ██░░░░░░░░░░██████░░░░░░░░░░██
+ //                           ██░░░░░░░░░░░░██████░░░░░░░░░░░░██
+ //                           ██░░░░░░░░░░░░██████░░░░░░░░░░░░██
+ //                         ██░░░░░░░░░░░░░░██████░░░░░░░░░░░░░░██
+ //                         ██░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░██
+ //                       ██░░░░░░░░░░░░░░░░██████░░░░░░░░░░░░░░░░██
+ //                       ██░░░░░░░░░░░░░░░░██████░░░░░░░░░░░░░░░░██
+ //                     ██░░░░░░░░░░░░░░░░░░██████░░░░░░░░░░░░░░░░░░██
+ //                     ██░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░██
+ //                       ██████████████████████████████████████████
+ //
+ // If you happen to change the layout of this structure,
+ // this will lead to an AccessViolationException in netCore when using a more recent version of the nuget.
+ // If you need to modify the definition, create a new interface NativeCallTargetDefinition# that will be consumed by the native layer
+
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+    internal struct NativeCallTargetDefinition2
```

Comment:
Could this class be named with a more explicit naming? or maybe `NativeCallTargetDefinition` shold be renamed to `LegacyNativeCallTargetDefinition` ?

Reply:
True, but I've just followed the 
` // If you need to modify the definition, create a new interface NativeCallTargetDefinition# that will be consumed by the native layer`
directive set on the file...